src/transformers/modeling_flax_bert.py

```diff
@@ -417,3 +403,104 @@ def __call__(self, input_ids, attention_mask=None, token_type_ids=None, position
             jnp.array(token_type_ids, dtype="i4"),
             jnp.array(position_ids, dtype="i4"),
         )
+
+
+class FlaxBertPredictionHeadTransform(nn.Module):
+    @nn.compact
+    def __call__(self, hidden_states):
+        hidden_states = nn.Dense(hidden_states.shape[-1], name="dense")(hidden_states)
+        hidden_states = nn.elu(hidden_states)  # TODO: ACT2FN[config.hidden_act]
+        return FlaxBertLayerNorm(name="LayerNorm")(hidden_states)
+
+
+class FlaxBertLMPredictionHead(nn.Module):
+
+    vocab_size: int
+
+    @nn.compact
+    def __call__(self, hidden_states):
+        # TODO: The output weights are the same as the input embeddings, but there is
+        #   an output-only bias for each token.
+        #   Need a link between the two variables so that the bias is correctly
+        #   resized with `resize_token_embeddings`
+
+        hidden_states = FlaxBertPredictionHeadTransform(name="transform")(hidden_states)
+        hidden_states = nn.Dense(self.vocab_size, name="decoder")(hidden_states)
+        return hidden_states
+
+
+class FlaxBertOnlyMLMHead(nn.Module):
+    vocab_size: int
+    hidden_size: int
+    intermediate_size: int
+    head_size: int
+    num_heads: int
+    num_encoder_layers: int
+    type_vocab_size: int
+    max_length: int
+
+    @nn.compact
+    def __call__(self, input_ids, attention_mask=None, token_type_ids=None, position_ids=None, labels=None):
+        # Embedding
+        embeddings = FlaxBertEmbeddings(
+            self.vocab_size, self.hidden_size, self.type_vocab_size, self.max_length, name="embeddings"
+        )(input_ids, token_type_ids, position_ids, attention_mask)
+
+        # N stacked encoding layers
+        encoder = FlaxBertEncoder(
+            self.num_encoder_layers, self.num_heads, self.head_size, self.intermediate_size, name="encoder"
+        )(embeddings, attention_mask)
+
+        # Compute the prediction scores
+        pooled = FlaxBertPooler(name="pooler")(encoder)
+        logits = FlaxBertLMPredictionHead(vocab_size=self.vocab_size, name="predictions")(encoder)
+
+        return pooled, logits
+
+
+class FlaxBertForMaskedLM(FlaxBertModel):
+    def __init__(self, config: BertConfig, state: dict, seed: int = 0, **kwargs):
+        super().__init__(config, state, seed, **kwargs)
+        self.module = partial(
```

Comment:
I know I'm late to the party but can you also just return a module instance rather than a partial, e.g. `return FlaxBertOnlyMLMHead(vocab_size=...)`?

Reply:
er `self.module = FlaxBertOnlyMLMHead(vocab_size=..., ...)`

Reply:
Done 👍